src/prefect/orion/orchestration/core_policy.py

```diff
@@ -412,6 +474,18 @@ async def before_transition(
         proposed_state: Optional[states.State],
         context: OrchestrationContext,
     ) -> None:
+        if proposed_state.is_running() and context.parameters.get(
```

Comment:
This setup couples `RestartFlowRun` and `PreventTransitionsFromTerminalStates` in a really tight manner - they must both be in the orchestration policy and in the right order. It seems cleaner to have just one rule `HandleTerminalTransition` that allows restarts if appropriate and blocks transitions otherwise.

Reply:
sounds good! I've split them up into two rules: one for Tasks and one for Flows